kodilla-hibernate/src/test/java/com/kodilla/hibernate/manytomany/dao/CompanyDaoTestSuite.java

```diff
@@ -51,11 +57,84 @@ public void testSaveManyToMany(){
         Assert.assertNotEquals(0, dataMaestersId);
         Assert.assertNotEquals(0, greyMatterId);
 
-        //CleanUp
-        try {
-            companyDao.deleteAll();
+        //cleanup
+
+        try{
+            companyDao.delete(softwareMachine);
+        } catch (Exception e) {
+            //move on
+        }
+        try{
+            companyDao.delete(dataMaesters);
+        } catch (Exception e) {
+            //move on
+        }
+        try{
+            companyDao.delete(greyMatter);
         } catch (Exception e) {
-            //do nothing
+            //move on
         }
+
+
+    }
+
+    @Test
+    public void testGettingEmployeesByLastName() {
+        //Given
+        Employee johnSmith = new Employee("John", "Smith");
+        Employee stephanieSmith = new Employee("Stephanie", "Smith");
+        Employee lindaKovalsky = new Employee("Linda", "Kovalsky");
+
+        Company softwareMachine = new Company("Software Machine");
+        Company dataMaesters = new Company("Data Maesters");
+        Company greyMatter = new Company("Grey Matter");
+
+        softwareMachine.getEmployees().add(johnSmith);
+        dataMaesters.getEmployees().add(stephanieSmith);
+        dataMaesters.getEmployees().add(lindaKovalsky);
+        greyMatter.getEmployees().add(johnSmith);
+        greyMatter.getEmployees().add(lindaKovalsky);
+
+        johnSmith.getCompanies().add(softwareMachine);
+        johnSmith.getCompanies().add(greyMatter);
+        stephanieSmith.getCompanies().add(dataMaesters);
+        lindaKovalsky.getCompanies().add(dataMaesters);
+        lindaKovalsky.getCompanies().add(greyMatter);
+
+        //When
+        List<Employee> smithList = employeeDao.retrieveEmployeeByLastName("Smith");
+        int expectedSmithNumber =2;
+
+        //Then
+        Assert.assertEquals(expectedSmithNumber, smithList.size());
+
+
+        //cleanup
+        try{
+            companyDao.delete(softwareMachine);
+        } catch (Exception e) {
+            //move on
+        }
+        try{
+            companyDao.delete(dataMaesters);
+        } catch (Exception e) {
+            //move on
+        }
+        try{
+            companyDao.delete(greyMatter);
+        } catch (Exception e) {
+            //move on
+        }
+
+
+    }
```

Comment:
ten test jest niepoprawny. Nie ma w nim zapisywania Employee. Jest za to tworzenie company ktore jest tutaj zbedne. Test powinien wygladac np tak
```
@Test
    public void testGettingEmployeesByLastName() {
        //Given
        Employee johnSmith = new Employee("John", "Smith");
        Employee stephanieSmith = new Employee("Stephanie", "Smith");
        Employee lindaKovalsky = new Employee("Linda", "Kovalsky");

        employeeDao.saveAll(Arrays.asList(johnSmith, stephanieSmith, lindaKovalsky));
        //When
        List<Employee> smithList = employeeDao.retrieveEmployeeByLastName("Smith");
        int expectedSmithNumber =2;

        //Then
        Assert.assertEquals(expectedSmithNumber, smithList.size());


        //cleanup
        try{
            employeeDao.deleteAll();
        } catch (Exception e) {
            //move on
        }

    }
```

Reply:
* pamietaj prosze zeby w application.properties byla flaga
`spring.jpa.hibernate.ddl-auto=create`